File: Quivi.Backoffice.Api/Dtos/AcquirerConfiguration.cs
using Quivi.Domain.Entities.Charges;

namespace Quivi.Backoffice.Api.Dtos
{
    public class AcquirerConfiguration
    {
        public required string Id { get; init; }
        public ChargeMethod Method { get; init; }
        public ChargePartner Partner { get; init; }
        public bool IsActive { get; init; }
        public required IReadOnlyDictionary<ChargePartner, IReadOnlyDictionary<ChargeMethod, object>> Settings { get; init; }
    }
}

File: quivi.backoffice.react.webapp/src/hooks/api/Dtos/acquirerconfigurations/AcquirerConfiguration.ts
import { ChargeMethod } from "../ChargeMethod";
import { ChargePartner } from "./ChargePartner";

export interface AcquirerConfiguration {
    readonly id: string;
    readonly method: ChargeMethod;
    readonly partner: ChargePartner;
    readonly isActive: boolean;
    readonly settings: Record<string, Record<string, any>>;
}